### PR TITLE
chore(tracing): fix inferred base service for pytest command ran with --ddtrace optional argument

### PR DESCRIPTION
**Pull request by wconti27**
# Motiviation

Add special case for `--ddtrace` pytest argument, that was causing CI tests to have a different service name than local testing. CI testing adds the `--ddtrace` arg to the command. Previously the code would skip any args that started with `-` as well as the following arg, but that should not be the case for this arg.

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by Yun-Kim — APPROVED** [review]
This is just ensuring tests are aware of the inferred test service names right? Wasn't able to find anything else in this PR. Otherwise LGTM from team mlobs

**Comment by wconti27** [comment]
> What was special about the `mako` module in this context?

The change was to preserve current mako span naming behavior in the case that `DD_SERVICE` is unset, resulting in `config.service` being unset. Previously the service would default to mako's default service name, but with this change `config.service` is set with the inferred service name. If we didn't change the mako code it would use this new inferred service name always, which is breaking. Instead, now we use `trace_utils.int_service` to compute the service, and it coalesces correctly to the default value if our config.service is the inferred service name.

**Review by mabdinur — APPROVED** [review]
Do we expect this change to have any customer impact (ie do we need a release note?)

**Comment by emmettbutler** [comment]
@wconti27 is that true of any other integrations, or just mako?

**Comment by wconti27** [comment]
> Do we expect this change to have any customer impact (ie do we need a release note?)

@mabdinur The only customer impact should be limited to those using ddtrace with pytest via the following syntax `pytest --ddtrace ...`. If you think we need a release note for that I can add it.

**Review by wantsui — APPROVED** [review]

**Comment by romainkomorndatadog** [comment]
> The only customer impact should be limited to those using ddtrace with pytest via the following syntax pytest --ddtrace ...

FWIW, `pytest` has lots of other single-arg options (eg: `pytest -s path/to/my/tests`, or `-x`, etc). It feels a little weird to special-case `--ddtrace`.

**Review by gnufede — APPROVED** [review]

**Review by P403n1x87 — APPROVED** [review]

File: releasenotes/notes/fix-pytest-inferred-service-naming-5ba12e991d040a1b.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+fixes:
+  - |
+    Updates the inferred base service name algorithm to ensure that arguments following ``--ddtrace`` are no
```
Comment (wconti27) [inline releasenotes/notes/fix-pytest-inferred-service-naming-5ba12e991d040a1b.yaml:4]:
```suggestion
    ci_visibility: Updates the inferred base service name algorithm to ensure that arguments following ``--ddtrace`` are no 
```

**Review by emmettbutler — APPROVED** [review]

**Pull request merged**